pkg/transport/transport.go

```diff
@@ -121,6 +121,13 @@ func handleRoute(c echo.Context, s service.Service, logger log.Logger) error {
 		return c.String(500, err.Error())
 	}
 
+	if wm.Data == nil || wm.Version == "" || wm.GroupKey == "" {
+		err = fmt.Errorf("validation of webhook message failed. Required fields are missing")
+		logger.Log("err", err)
```

Comment:
Maybe we should also add a debug log with more information what fields are missing?

Reply:
The issue is that "Data" is not a field of the json itself. 
If any field of template.Data is set, Data will not be empty and no panic will happen.
Any field of this list is enough: https://github.com/prometheus/alertmanager/blob/v0.20.0/template/template.go#L231-L241.

Version and GroupKey are not needed to prevent panic. I just enforced a normal webhook message format here.

I could change it to be more explicit/minimal:
```
if wm.Data == nil {
		err = fmt.Errorf("validation of webhook message failed. Fields receiver, status, alerts,... are missing")
                logger.Log("err", err)
 
```

Reply:
I see... Then let's rephrase the error message a little bit, maybe something like: "The webhook message does not seem to be a valid Prometheus Alertmanager webhook. More information see https://prometheus.io/docs/alerting/latest/configuration/#webhook_config".
What do you think? Then a user knows how to further debug.

Reply:
@Knappek  I've used your proposal. I agree it's more helpful for the end user.